src/collection.js

```diff
@@ -37,106 +69,189 @@ CKEDITOR.define( [ 'emittermixin', 'ckeditorerror', 'utils' ], function( Emitter
 		 * @property length
 		 */
 		get length() {
-			return this._models.length;
+			return this._items.length;
 		}
 
 		/**
 		 * Adds an item into the collection.
 		 *
-		 * Note that this is an array-like collection, so the same item can be present more than once. This behavior is
-		 * for performance purposes only and is not guaranteed to be kept in the same way in the future.
+		 * If the item does not have an id, then it will be automatically generated and set on the item.
 		 *
-		 * @param {Model} model The item to be added.
+		 * @chainable
+		 * @param {Object} item
 		 */
-		add( model ) {
-			this._models.push( model );
+		add( item ) {
+			var itemId;
+			var idProperty = this._idProperty;
+
+			if ( ( idProperty in item ) ) {
+				itemId = item[ idProperty ];
+
+				if ( typeof itemId != 'string' ) {
```

Comment:
Do we need such limitation? Why "symbol" can not be a key? We already have a case in the `Document` where we have a collection of root elements and the root of graveyard is kept by a symbol, so it will not conflict with any other name of the root.